sources/packages/backend/apps/api/src/services/disbursement-receipt/disbursement-receipt.service.ts

```diff
@@ -54,4 +72,31 @@ export class DisbursementReceiptService extends RecordDataModelService<Disbursem
       },
     });
   }
+
+  /**
+   * Get the related loan award code based in the receipt.
+   * Provincial loans are not present for part-time disbursements.
+   * @param fundingType defines if the receipt is for a federal or provincial
+   * part of the disbursement.
+   * @param offeringIntensity offering intensity associate with the assessment
+   * related to the receipt.
+   * @returns loan award code, if possible.
+   */
+  static getLoanAwardCode(
+    fundingType: string,
+    offeringIntensity: OfferingIntensity,
+  ): "CSLF" | "BCSL" | "CSLP" | null {
+    if (offeringIntensity === OfferingIntensity.fullTime) {
```

Comment:
Just a suggestion, instead of if and if a switch case will give better readability and default null. :)

Reply:
That would be a switch case with two conditions which, in my opinion, would not make the code easier to read. Open for discussion but if you do belive that would be better, please paste a sample code for my better understanding, otherwise I will leave as it is 😉

Reply:
It was just a suggestion, i was saying something like this

static getLoanAwardCode(
  fundingType: string,
  offeringIntensity: OfferingIntensity,
): "CSLF" | "BCSL" | "CSLP" | null {
  switch (offeringIntensity) {
    case OfferingIntensity.fullTime:
      -------------
    case OfferingIntensity.partTime:
      ------------
    default:
      return null;
  }
}